### PR TITLE
Doctest fix I

### PR DESCRIPTION
**Pull request by bsipocz**
As mentioned in #1177 there are 236 failing doctests of the 434. There are a mixture of genuine fails and trivial ones where the lines should be just simply skipped.

E.g. the numbers in ``database.rst`` changed genuinely as a result of additional observations were added to the table it queries since the original docs were written. It also failed at several places due to trivial whitespace differences. I could only deal with it using ``+NORMALIZE_WHITESPACE``.

This PR fixes the ones in the docs (but not the ones in the docstrings).

Relies on #1452 and #1453.

### CODE REVIEW

**Comment by Cadair** [comment]
:+1:

**Comment by bsipocz** [comment]
@Cadair, @ehsteve - Which way do you prefer to proceed with the doctests? Should I push more commit to this PR, or open new ones? I guess the latter would be easier to review, but the former means less noise.

**Comment by Cadair** [comment]
@bsipocz the advantage of doing it all in one go is that you can enable the doctests on Travis and check that they all pass. However, you can always do that on the last PR of a set as well.

I do not mind which one you pick. Splitting it up would probably be preferable.

**Comment by bsipocz** [comment]
@Cadair - these new commits should fix all doctests. However some of the remote ones, e.g. net/jsoc and net/vso moduls produce sporadic failures, they were working fine yesterday night, but not today. 

I've also added the doctest run to the travis matrix, but allow it to fail for the time being.

There are still a load of examples that are not picked up by doctest (they doesn't start with ``>>>`` or the directive), they are not fixed in this PR.

**Comment by bsipocz** [comment]
I don't get why it failed in the database module and docs, at the moment locally all but two vso tests pass.

```
Document: guide/acquiring_data/database
---------------------------------------
1 items passed all tests:
  68 tests in default
68 tests in 1 items.
68 passed and 0 failed.
Test passed.
```

```
Doctest summary
===============
  421 tests
    2 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems.
```

**Comment by bsipocz** [comment]
Rebased on master as the major #1432 is merged now. Every doctest passes locally.

**Comment by bsipocz** [comment]
Strangely the database.rst outputs are differently sorted when testing them locally (on a Mac), and on travis or other Linux box. I don't quite understand why the order of the database entries change. Any idea? 
When is your 0.6 due? After a review this may be ready for merge as an improvement to the docs.

File: .travis.yml [hunk 1]
```diff
@@ -37,10 +37,13 @@ matrix:
            env: TEST_MODE='sphinx'
          - os: linux
            env: TEST_MODE='online'
+         - os: linux
+           env: TEST_MODE='doctest'
 
 # If the online build fails, we don't want it to report the whole build broken.
     allow_failures:
         - env: TEST_MODE='online'
+        - env: TEST_MODE='doctest'
```
Comment (Cadair) [inline .travis.yml]: Did you say they were all passing? If so we can move this out of here.

**Comment by Cadair** [comment]
I don't know why the ordering of the database outputs would change, or more over what to do about it. @derdon any ideas??

**Comment by Cadair** [comment]
@sunpy/sunpy-maintainers review on this please.

File: .travis.yml (hunk 1 above)
Comment (bsipocz) [inline .travis.yml]: Yes, they passed locally, but not on Travis. The 13 one still fails there.
Also some of  the tests use online data, so I'm not sure whether requiring
all do tests to pass is a good idea.

Comment (Cadair) [inline .travis.yml]: Ah right OK. Makes sense.

**Comment by ehsteve** [comment]
btw I am working on adding the plot directive to the docs.

**Comment by bsipocz** [comment]
@ehsteve - :1 to that. I think (but not 100% sure) that the doctest skips the codes in the plot directive anyway.
I also have another branch to change and fix the docs examples so doctest can pick them up.

**Comment by Cadair** [comment]
While we are in this, I am working on making the little hide >>> JavaScript work for the code examples. So please stop removing them ;)

**Comment by bsipocz** [comment]
I've put the copybutton.js into the astropy template for this, and tried to
make it work here too, but it's not working yet.
Anyway, when I say that I'm changing the examples means exactly to put
``>>>`` back, otherwise doctest doesn't recognize it as code.

**Comment by Cadair** [comment]
@bsipocz yeah, I will continue trying to get the `copybutton.js` thing to work, my first issue was with jQuery, and now I am having an issue with the whole div being disabled for some reason I can not work out.

Also, sorry if I wasn't clear with my comment yesterday, I sent it from my phone in a rush, I was telling @ehsteve to stop removing the `>>>` as I thought they were needed for the doctests and I was working on copybutton.

**Comment by ehsteve** [comment]
:+1:

**Comment by Cadair** [comment]
:+1:

**Comment by Cadair** [comment]
@bsipocz We decided to merge this so that we can continue to improve the docs for 0.6. Please feel free to continue fixing up doctests.

**Pull request merged**